source/common/upstream/load_balancer_impl.cc

```diff
@@ -774,6 +780,10 @@ void EdfLoadBalancerBase::refresh(uint32_t priority) {
 }
 
 HostConstSharedPtr EdfLoadBalancerBase::peekAnotherHost(LoadBalancerContext* context) {
+  if (stashed_random_.size() >= total_healthy_hosts_) {
```

Comment:
Some comments? Not immediately obvious to me what this is doing. Same elsewhere. Possibly some well named helper might make it easier to read.